src/ast-parser.y

```diff
@@ -1112,12 +1149,15 @@ inline_export :
 
 type_def :
     LPAR TYPE func_type RPAR {
-      WABT_ZERO_MEMORY($$);
-      $$.sig = $3;
+      $$ = new FuncType();
+      $$->sig = std::move(*$3);
+      delete $3;
     }
   | LPAR TYPE bind_var func_type RPAR {
-      $$.name = $3;
-      $$.sig = $4;
+      $$ = new FuncType();
```

Comment:
nit: Alternatively, you could define separate constructors that take these arguments explicitly, reducing the code to

  $$ = new FuncType($3, $4)

This will also allow the delete to be inside the construtor.